### PR TITLE
Feature - Add new testimonial component

### PR DESCRIPTION
**Pull request by HarshArora-1205**
https://github.com/user-attachments/assets/b8ebc098-d7bc-46cf-9e1a-3d25812c167c

### CODE REVIEW

File: package-lock.json
Comment (markmead) [inline package-lock.json:1]: Delete this file, please.

File: yarn.lock
Comment (markmead) [inline yarn.lock:1]: Shouldn't be any changes in this file.

File: public/components/marketing-testimonials/6.html [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+<script type="module"> 
+  import KeenSlider from 'https://cdn.jsdelivr.net/npm/keen-slider@6.8.6/+esm';
+  
+  const keenSlider = new KeenSlider(
+    '#keen-slider',
+    {
+      loop: true,
+
+      // Change rtl to true for reversing slides arrangement
+      rtl: false,
```
Comment (markmead) [inline public/components/marketing-testimonials/6.html:12]: Can delete this.

File: public/components/marketing-testimonials/6.html [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+      slides: {
+        origin: 'center',
+        perView: 1,
+        spacing: 16,
+      },
+      // Slide Scroll Speed
```
Comment (markmead) [inline public/components/marketing-testimonials/6.html:18]: No need for the comment.

File: public/components/marketing-testimonials/6.html [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+      defaultAnimation: {
+        duration: 1500,
+      },
+      breakpoints: {
```
Comment (markmead) [inline public/components/marketing-testimonials/6.html:22]: Breakpoints are missing closing `)`.

File: public/components/marketing-testimonials/6.html [hunk 4]
```diff
@@ -0,0 +1,436 @@
...
+      // Set new slide opacity to 1 on carousel slide change
+      slideChanged: (slider) => {
+        const activeSlide = slider.track.details.rel;
+        
+        slider.slides.forEach((element, idx) => {
+          element.style.opacity = 0.4;
+        });
+        slider.slides[activeSlide].style.opacity = 1;
+      },
+      renderMode: 'performance',
```
Comment (markmead) [inline public/components/marketing-testimonials/6.html:72]: Don't think we need this.

**Review by markmead — COMMENTED** [review]
Looks good! Instead of adding/removing `style="opacity: 0.4"` you could add `opacity-40` class to all the slides by default, then toggle that class for the active slide.

File: yarn.lock
Comment (HarshArora-1205) [reply yarn.lock:1]: Restored previous version

File: public/components/marketing-testimonials/6.html (hunk 1 above)
Comment (HarshArora-1205) [reply public/components/marketing-testimonials/6.html]: addressed.

File: public/components/marketing-testimonials/6.html (hunk 4 above)
Comment (HarshArora-1205) [reply public/components/marketing-testimonials/6.html]: removed

File: public/components/marketing-testimonials/6.html (hunk 3 above)
Comment (HarshArora-1205) [reply public/components/marketing-testimonials/6.html:18]: Resolved.

File: public/components/marketing-testimonials/6.html (hunk 2 above)
Comment (HarshArora-1205) [reply public/components/marketing-testimonials/6.html]: Resolved.

**Comment by HarshArora-1205** [comment]
> Looks good! Instead of adding/removing `style="opacity: 0.4"` you could add `opacity-40` class to all the slides by default, then toggle that class for the active slide.

Added that part, but toggling only active slide won't help. We have to somehow toggle previous slide also. Keen slider does not explicitly provide us any previous slide index for that. 
So I wrote a logic to add a previousSlide option directly to the slider object. This workaround works fine but in case it breaks the solution would be the same, that is, to toggle on for all slides for class (opacity-40).

**Comment by HarshArora-1205** [comment]
> PR looks good, I'll pull down the work in a bit as I want to play with `keen-slider` a little.

Cool. Ping me if you find some better solution. Fetching active slide is easier though but toggling classes would require a loop on all slides or we have to store previous slide somewhere.

By the way can we have a image carousel collection in HyperUI 🤔? If so, I would be happy to contribute.

> Bugs me there's not a way to style the slides in the HTML.

Yep. That's where embla's plugin made it easier. To just toggle opacity when there is .is-snapped classname on a slide.

```html
<div class="embla__slide group">
  <blockquote class="opacity-40 group-[.is-snapped]:opacity:100">
  
  </blockquote>
</div>
```


I think slick carousel library also uses classnames by default, to label prev curr and next slide.

**Comment by markmead** [comment]
Made some changes @HarshArora-1205 as I wasn't a big fan of the navigation buttons being on top of the slides. Sorry, I didn't mention this earlier, didn't come to that conclusion until I pulled down your work and started poking around.

Let me know if you're OK with this and if so, I'll get it merged.

**Comment by HarshArora-1205** [comment]
>Let me know if you're OK with this and if so, I'll get it merged.

Looks good. Handles mobile view better in terms of user experience. 

Further, not explicitly tracking previous slide makes it simpler. (for limited reviews)

**Comment by markmead** [comment]
> Further, not explicitly tracking previous slide makes it simpler. (for limited reviews)

Yeah, I was thinking keep it simple as not everyone will end up using `keen-slider`, so not point sweating over the details.

**Pull request merged**